packages/cells/src/widget.ts

```diff
@@ -1603,6 +1604,14 @@ export class CodeCell extends Cell<ICodeCellModel> {
     }
     // This is to hide/show icon on single line output.
     this.updatePromptOverlayIcon();
+
+    // Clear output area when empty
+    this._lastOutputHeight = this.outputArea.node.style.height;
+    if (this.model.outputs.length === 0) {
+      this.outputArea.node.style.height = '';
```

Comment:
I think the assignment needs to be inside the conditional as otherwise we would overwrite the cached value with `''` set after it was stored.

```suggestion
    if (this.model.outputs.length === 0) {
      this._lastOutputHeight = this.outputArea.node.style.height;
      this.outputArea.node.style.height = '';
```